.github/workflows/haskell-ci.yml

```diff
@@ -301,8 +326,24 @@ jobs:
           ${CABAL} -vnormal check
       - name: haddock
         run: |
-          if [ $((HCNUMVER < 80200 || HCNUMVER >= 80400)) -ne 0 ] ; then $CABAL v2-haddock --haddock-all $ARG_COMPILER --with-haddock $HADDOCK $ARG_TESTS $ARG_BENCH all ; fi
+          if [ $((HCNUMVER < 80200 || HCNUMVER >= 80400)) -ne 0 ] ; then $CABAL v2-haddock --disable-documentation --haddock-all $ARG_COMPILER --with-haddock $HADDOCK $ARG_TESTS $ARG_BENCH all ; fi
       - name: unconstrained build
         run: |
           rm -f cabal.project.local
           $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks all
+      - name: prepare for constraint sets
+        run: |
+          rm -f cabal.project.local
+      - name: constraint set text-2.1
+        run: |
+          if [ $((HCNUMVER >= 80200)) -ne 0 ] ; then $CABAL v2-build $ARG_COMPILER --enable-tests --disable-benchmarks --constraint='text ^>= 2.1' all --dry-run ; fi
+          if [ $((HCNUMVER >= 80200)) -ne 0 ] ; then cabal-plan topo | sort ; fi
+          if [ $((HCNUMVER >= 80200)) -ne 0 ] ; then $CABAL v2-build $ARG_COMPILER --enable-tests --disable-benchmarks --constraint='text ^>= 2.1' --dependencies-only -j2 all ; fi
+          if [ $((HCNUMVER >= 80200)) -ne 0 ] ; then $CABAL v2-build $ARG_COMPILER --enable-tests --disable-benchmarks --constraint='text ^>= 2.1' all ; fi
+          if [ $((HCNUMVER >= 80200)) -ne 0 ] ; then $CABAL v2-test $ARG_COMPILER --enable-tests --disable-benchmarks --constraint='text ^>= 2.1' all ; fi
```

Comment:
Ditto here, what are these lines for? Does this file borrow from [this file](https://github.com/haskell-CI/haskell-ci/blob/master/fixtures/all-versions.github) (I don't see these lines there) or the CLI from that project or neither?

Reply:
Can you include the CLI used to generate it as a comment at the top of the file, or otherwise in the PR itself? If it's generated, it will be useful to know what the command was.

Reply:
Haskell CI puts it there see: https://github.com/stevenfontanella/microlens/blob/4eb1617ddac7fb74eb36d78ee29cde8bc302b079/.github/workflows/haskell-ci.yml#L1-L7